### DISCUSSION TITLE
[Question] Could materialized view work for historical data?

### DISCUSSION THREAD

**Discussion by sundy-li** [risingwavelabs/risingwave#1732] [Q&A]
Hi, I have some doubts after I read the design document.

1. Could materialized view work for historical data?
2. If it could, so it relies on the sourcer(kafka,mq) to reset the offset to the earliest?

**Comment by TennyZhuang** [comment]
1. Yes, if the offset of source can be subscribed.
2. Another option is creating a materialized source, sorry for that it’s not well documented, but in fact it’s an materialized view behind a source. Then while you create a new mview based on the materialized source, the historical data can be recovered, see https://github.com/singularity-data/risingwave/blob/main/docs/mv-on-mv.md for more information.
